x/auth/types/credentials.go

```diff
@@ -44,7 +44,12 @@ func NewModuleCredential(moduleName string, moduleAccountKey, derivationKey []by
 }
 
 func (m *ModuleCredential) Address() cryptotypes.Address {
-	return address.Derive(address.Module(m.ModuleName, m.ModuleAccountKey), m.DerivationKey)
+	module := address.Module(m.ModuleName, m.ModuleAccountKey)
```

Comment:
Reading ADR-028, I believe it's:

```suggestion
	if m.ModuleAccountKey != nil {
		module := address.Module(m.ModuleName, m.ModuleAccountKey)
	} else {
		module := address.Hash("module", moduleName)
	}
```